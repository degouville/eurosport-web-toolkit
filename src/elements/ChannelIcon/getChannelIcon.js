import { ReactComponent as EComponent } from '../../assets/channels/E.svg';
import { ReactComponent as E1Component } from '../../assets/channels/E1.svg';
import { ReactComponent as E2Component } from '../../assets/channels/E2.svg';
import { ReactComponent as E2GRComponent } from '../../assets/channels/E2GR.svg';
import { ReactComponent as E2NOComponent } from '../../assets/channels/E2NO.svg';
import { ReactComponent as E2RUGComponent } from '../../assets/channels/E2RUG.svg';

export const icons = {
  E2GR: {
    altText: 'Eurosport 2',
    widthRatio: 3.5,
    component: E2GRComponent,
  },
  E2NO: {
    altText: 'Eurosport 2',
    widthRatio: 2.06,
    component: E2NOComponent,
  },
  E2RUG: {
    altText: 'Eurosport 2',
    widthRatio: 4,
    component: E2RUGComponent,
  },
  E1: {
    altText: 'Eurosport 1',
    widthRatio: 1.42,
    component: E1Component,
  },
  E2: {
    altText: 'Eurosport 2',
    widthRatio: 1.66,
    component: E2Component,
  },
  E: {
    altText: 'Eurosport',
    widthRatio: 0.87,
    component: EComponent,
  },
};

export default callsign => {
  if (callsign === 'E2GR') {
    return icons.E2GR;
  }

  if (callsign === 'E2NO') {
    return icons.E2NO;
  }

  if (callsign === 'E2RUG') {
    return icons.E2RUG;
  }

  if (callsign.startsWith('E1')) {
    return icons.E1;
  }

  if (callsign.startsWith('E2')) {
    return icons.E2;
  }
  return icons.E;
};
