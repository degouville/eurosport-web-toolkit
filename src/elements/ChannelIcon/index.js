import React from 'react';
import PropTypes from 'prop-types';
import E from '../../assets/channels/E.svg';
import E1 from '../../assets/channels/E1.svg';
import E2 from '../../assets/channels/E2.svg';
import E2GR from '../../assets/channels/E2GR.svg';
import E2NO from '../../assets/channels/E2NO.svg';
import E2RUG from '../../assets/channels/E2RUG.svg';

export const iconMap = {
  E: {
    src: E,
    altText: 'Eurosport',
    widthRatio: 0.87,
  },
  E1: {
    src: E1,
    altText: 'Eurosport 1',
    widthRatio: 1.42,
  },
  E2: {
    src: E2,
    altText: 'Eurosport 2',
    widthRatio: 1.66,
  },
  E2GR: {
    src: E2GR,
    altText: 'Eurosport 2',
    widthRatio: 3.5,
  },
  E2NO: {
    src: E2NO,
    altText: 'Eurosport 2',
    widthRatio: 2.06,
  },
  E2RUG: {
    src: E2RUG,
    altText: 'Eurosport 2',
    widthRatio: 4,
  },
};

const ChannelIcon = ({ type, height, ...props }) => {
  const icon = iconMap[type] || iconMap.E;

  return icon ? (
    <img
      {...props}
      css={`
        height: ${height}px;
        width: ${icon.widthRatio * height}px;
      `}
      src={icon.src}
      alt={icon.altText}
    />
  ) : null;
};

ChannelIcon.defaultProps = {
  type: 'E',
  height: 50,
};

ChannelIcon.propTypes = {
  type: PropTypes.oneOf(['E', 'E1', 'E2', 'E2NO', 'E2RUG', 'E2GR']),
  height: PropTypes.number,
};

export default ChannelIcon;
