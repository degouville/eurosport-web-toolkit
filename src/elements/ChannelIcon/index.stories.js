import React from 'react';
import { storiesOf } from '@storybook/react';
import { number } from '@storybook/addon-knobs';
import { ChannelIcon } from '../..';

const iconsStories = storiesOf('Elements|ChannelIcon', module);

const iconTypes = ['', 'E1*', 'E2*', 'E2NO', 'E2RUG', 'E2GR'];

iconTypes.forEach(type => {
  iconsStories.add(`ChannelIcon${type ? ` - ${type}` : ''}`, () => (
    <ChannelIcon type={type} height={number('Height', 50)} />
  ));
});
