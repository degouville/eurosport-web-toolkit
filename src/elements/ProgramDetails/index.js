import React from 'react';
import styled, { css } from 'react-emotion';
import PropTypes from 'prop-types';
import ChannelIcon from '../ChannelIcon';
import { coreNeutral4 } from '../../colors';
import { medium } from '../../breakpoints';

export const StyledDetails = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  height: 100%;
  border-top: 1px solid rgba(255, 255, 255, 0.15);
  color: ${coreNeutral4};
  font-size: 11px;

  svg {
    width: auto;
    height: 100%;
  }

  ${medium(css`
    font-size: 14px;
    line-height: 20px;
  `)};
`;

export const StyledIconWrapper = styled.div`
  height: 15px;
  padding: 10px 0;

  ${medium(css`
    padding: 15px 0;
    height: 23px;
  `)};
`;

export const StyledText = styled.div`
  padding: 10px 10px 10px 0;

  ${medium(css`
    padding: 15px 15px 15px 0;
  `)};
`;

export const StyledSeparator = styled.div`
  width: 1px;
  background: rgba(255, 255, 255, 0.15);
  margin: 0 10px;
  align-self: stretch;
  flex-shrink: 0;
`;

const ProgramDetails = ({ customIcon, callsign, textDetail, className }) => (
  <StyledDetails className={className}>
    {callsign && (
      <>
        <StyledIconWrapper>
          <ChannelIcon type={callsign} />
        </StyledIconWrapper>
        <StyledSeparator />
      </>
    )}
    {!callsign && customIcon && (
      <>
        {customIcon}
        <StyledSeparator />
      </>
    )}
    <StyledText>{textDetail}</StyledText>
  </StyledDetails>
);

ProgramDetails.defaultProps = {
  callsign: 'E',
  className: undefined,
  customIcon: null,
};

ProgramDetails.propTypes = {
  className: PropTypes.string,
  callsign: PropTypes.string,
  customIcon: PropTypes.node,
  textDetail: PropTypes.string.isRequired,
};

export default ProgramDetails;
