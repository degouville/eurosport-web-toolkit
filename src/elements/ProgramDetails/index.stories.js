import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import ProgramDetails from '.';

const ProgramDetailsStories = storiesOf('Elements|ProgramDetails', module);

ProgramDetailsStories.add('default', () => (
  <ProgramDetails callsign={text('callsign', 'E')} textDetail={text('textDetail', '09:00 - 10:00')} />
));
