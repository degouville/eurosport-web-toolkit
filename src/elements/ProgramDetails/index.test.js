import React from 'react';
import { shallow, mount } from 'enzyme';
import ProgramDetails from '.';
import ChannelIcon from '../ChannelIcon';

describe.only('PlayIcon', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<ProgramDetails textDetail="Text details" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the default channel icon', () => {
    const wrapper = mount(<ProgramDetails textDetail="Text details" />);
    expect(wrapper.find(ChannelIcon).prop('type')).toBe('E');
  });

  it('should not render a channel icon', () => {
    const wrapper = mount(<ProgramDetails textDetail="Text details" callsign="" />);
    expect(wrapper.find(ChannelIcon)).toHaveLength(0);
  });
});
