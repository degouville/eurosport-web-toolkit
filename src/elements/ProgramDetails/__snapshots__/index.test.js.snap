// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Program details should render as expected 1`] = `
<Styled(div)
  className=""
>
  <Styled(div)>
    <ChannelIcon
      height={50}
      type="E2NO"
    />
  </Styled(div)>
  <Styled(div) />
  <Styled(div)>
    Text details
  </Styled(div)>
</Styled(div)>
`;
